aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java

```diff
@@ -66,21 +67,32 @@ public class AwsClientProperties implements Serializable {
    */
   public static final String CLIENT_REGION = "client.region";
 
+  /**
+   * Configure the endpoint to be used to fetch and refresh vended credentials.
+   *
+   * <p>When set, the {@link VendedCredentialsProvider} will be used to fetch and refresh vended
+   * credentials.
+   */
+  public static final String REFRESH_CREDENTIALS_ENDPOINT = "client.refresh-credentials-endpoint";
```

Comment:
we most likely also need a property to explicitly disable refreshing vended creds for Kafka Connect